netutils/dhcpc/dhcpc.c

```diff
@@ -665,6 +666,17 @@ int dhcpc_request(FAR void *handle, FAR struct dhcpc_state *presult)
   int     retries;
   int     state;
 
+  /* RFC2131: A DHCP client MUST choose 'xid's in such a
+   * way as to minimize the chance of using an 'xid' identical to one used
+   * by another client.
+   */
+
+  ret = getrandom(&xid, sizeof(xid), 0);
```

Comment:
move to dhcpc_open

Reply:
I think we should put it in dhcp request, so that if we keep the dhcpc thread, xid will change every time just like the rfc requested.

Reply:
here is my suggestion:
1.Move xid to dhcpc_state_s
2.Initialize xid with random number in dhcp_open
3.update inside dhcp_request